chain/ethereum/src/chain.rs

```diff
@@ -763,23 +819,96 @@ impl TriggersAdapterTrait<Chain> for TriggersAdapter {
         logger: Logger,
         block_numbers: HashSet<BlockNumber>,
     ) -> Result<Vec<BlockFinality>> {
-        use graph::futures01::stream::Stream;
+        // Common function to handle block loading, regardless of source
+        async fn load_blocks<F, Fut>(
```

Comment:
is there a reason this couldn't be a function on triggers adapter instead of inside the function? This is more of a nitpick but this usually means it's code that cannot be tested and personally I think it makes code less readable. I understand the code locality here and I think it does help understand the full context (in this case). Just wary of doing this sort of thing as a norm

Reply:
The reason is that its a very specific function and it is not at all likely to be used anywhere else since this method takes closure which `fetch_missing`. Overall its a very specific function and wouldn't be used anywhere else. But yes feels like this could be moved outside, I'll just go with an external function, instead of inline